app/src/lib/editors/win32.ts

```diff
@@ -0,0 +1,253 @@
+    if (!exists) {
+      log.debug(`Command line interface for ${name} not found at '${path}'`)
+      return {
+        name,
+        installed: true,
+        pathExists: false,
+      }
+    } else {
+      return {
+        name,
+        installed: true,
+        pathExists: true,
+        path,
+      }
+    }
+  } catch (error) {
+    log.debug(`Unable to locate ${name} installation`, error)
+    return { name, installed: false }
+  }
+}
+
+/**
+ * Find the Visual Studio Code executable shim using the install information
+ * from the registry.
+ */
+async function findCodeApplication(): Promise<LookupResult> {
+  const name = VisualStudioCodeLabel
+
+  try {
+    const keys = await readKeys(
+      'HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{F8A2A208-72B3-4D61-95FC-8A65D340689B}_is1'
+    )
+
+    let displayName = ''
+    let publisher = ''
+    let installLocation = ''
+
+    for (const item of keys) {
+      if (item.name === 'Inno Setup: Icon Group') {
+        displayName = item.value
+      } else if (item.name === 'Publisher') {
+        publisher = item.value
+      } else if (item.name === 'Inno Setup: App Path') {
+        installLocation = item.value
+      }
+    }
+
+    const isExpectedInstall =
+      displayName === 'Visual Studio Code' &&
+      publisher === 'Microsoft Corporation'
+
+    if (!isExpectedInstall) {
+      log.debug(
+        `Registry entry for ${name} did not match expected publisher settings`
+      )
+      return {
+        name,
+        installed: true,
+        pathExists: false,
+      }
+    }
+
+    const path = Path.join(installLocation, 'bin', 'code.cmd')
+    const exists = await pathExists(path)
+    if (!exists) {
+      log.debug(`Command line interface for ${name} not found at '${path}'`)
+      return {
+        name,
+        installed: true,
+        pathExists: false,
+      }
+    } else {
+      return {
+        name,
+        installed: true,
+        pathExists: true,
+        path,
+      }
+    }
+  } catch (error) {
+    log.debug(`Unable to locate ${name} installation`, error)
+    return { name, installed: false }
+  }
+}
+
+/**
+ * Find the Sublime Text executable shim using the install information from the
+ * registry.
+ */
+export async function findSublimeTextApplication(): Promise<LookupResult> {
+  const name = SublimeTextLabel
+
+  try {
+    const keys = await readKeys(
+      'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Sublime Text 3_is1'
+    )
+
+    let displayName = ''
+    let publisher = ''
+    let installLocation = ''
+
+    for (const item of keys) {
+      if (item.name === 'Inno Setup: Icon Group') {
+        displayName = item.value
+      } else if (item.name === 'Publisher') {
+        publisher = item.value
+      } else if (item.name === 'Inno Setup: App Path') {
+        installLocation = item.value
+      }
+    }
+
+    const expectedInstall =
```

Comment:
Same note as in the Darwin implementation, wondering if we can share this general pattern.